auth/auth.go

```diff
@@ -27,32 +30,48 @@ import (
 )
 
 const (
-	arnAnno       = "eks.amazonaws.com/role-arn"
-	docURL        = "https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html"
-	tokenAudience = "sts.amazonaws.com"
-	ProviderName  = "secrets-store-csi-driver-provider-aws"
+	arnAnno                         = "eks.amazonaws.com/role-arn"
+	docURL                          = "https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html"
+	irsaAudience                    = "sts.amazonaws.com"
+	podIdentityAudience             = "pods.eks.amazonaws.com"
+	ProviderName                    = "secrets-store-csi-driver-provider-aws"
+	defaultPodIdentityAgentEndpoint = "http://169.254.170.23/v1/credentials"
```

Comment:
The default pod identity agent IPV6 address is `fd00:ec2::23`, can we support IPV6 as long as the customer hasn't disabled IPV6 in their cluster?

https://docs.aws.amazon.com/eks/latest/userguide/pod-identities.html#pod-id-considerations